SimPEG/regularization/pgi.py

```diff
@@ -610,20 +680,27 @@ def W(self):
         Weighting matrix
         Need to change the size to match self.wiresmap.maps * mesh.nC
         """
+
         if self.cell_weights is not None:
             if len(self.cell_weights) == self.wiresmap.nP:
-                return sp.kron(
-                    speye(len(self.wiresmap.maps)), sdiag(np.sqrt(self.regmesh.vol)),
-                ) * sdiag(np.sqrt(self.cell_weights))
+                return (
+                    sp.kron(
+                        speye(len(self.wiresmap.maps)),
+                        sdiag(np.sqrt(self.regmesh.vol)),
+                    )
+                    * sdiag(np.sqrt(self.cell_weights))
+                )
             else:
                 return sp.kron(
-                    speye(len(self.wiresmap.maps)), sdiag(np.sqrt(self.regmesh.vol)),
+                    speye(len(self.wiresmap.maps)),
+                    sdiag(np.sqrt(self.regmesh.vol)),
```

Comment:
`sp.kron` is used here with `len(self.wiresmap.maps)` for cases with multiple physical properties. `W` is then a matrix of (n*p) X (n*p) size (n:number of active cells, p: number of physical properties) with repeated volumes weights for each phys. prop (thus, the `sp.kron(speye(len(self.wiresmap.maps)), [...]`.

There might be a better way; we could form the diagonal with a nested loop `[weights for I in range(len(..)]` and then call `sdiag` but would that be a significant gain?

Reply:
It would certainly make the code mode readable with less operations. Plus you can avoid the if statement by always kroning your vector with len(self.wiresmap.maps)

Reply:
It would certainly make the code mode readable with less operations. Plus you can avoid the if statement by always kroning your vector with len(self.wiresmap.maps)